docs/ref/models/database-functions.txt

```diff
@@ -1364,6 +1589,35 @@ Usage example::
     >>> print(author.name_md5)
     749fb689816b2db85f5b169c2055b247
 
+``OctetLength``
```

Comment:
On MariaDB `OCTET_LENGTH` is a synonym for `LENGTH`, except in Oracle mode https://mariadb.com/kb/en/octet_length/

If this is an Oracle specific thing, I'm not sure it's worth adding in the genral function library. Perhaps it should be in `django.contrib.oracle`.

Also perhaps it should be called `ByteLength`. It seems we don't use the name octet anywhere in Python except the mime type.

Reply:
> On MariaDB `OCTET_LENGTH` is a synonym for `LENGTH`, except in Oracle mode https://mariadb.com/kb/en/octet_length/

So the ANSI SQL standard functions are `BIT_LENGTH`, `CHAR_LENGTH` and `OCTET_LENGTH` -- see [here](https://www.oreilly.com/library/view/sql-in-a/9780596155322/re81.html). Most backends support `LENGTH` as a synonym for `CHAR_LENGTH`. We explicitly use `CHAR_LENGTH` for MySQL, and thus MariaDB, which will avoid the issue of Oracle mode -- see [here](https://github.com/django/django/blob/eb215da363e6cf0e8f3405db3c4392398c8777cb/django/db/models/functions/text.py#L150-L151). I could add a commit to prefer `CHAR_LENGTH` over `LENGTH` for `Length()` where possible if desired.

> If this is an Oracle specific thing, I'm not sure it's worth adding in the genral function library. Perhaps it should be in `django.contrib.oracle`.

So, this is not Oracle-specific.

> Also perhaps it should be called `ByteLength`. It seems we don't use the name octet anywhere in Python except the mime type.

I thought of this but wasn't sure. Now that you've raised it too, I think it would be more accesible.

Reply:
We could add `BitLength` at the same time if it's ANSI - or at least make a ticket for it!

Reply:
Umm. I have done... :confused: :joy:

Reply:
My bad 🙈